java/ql/src/experimental/Security/CWE/CWE-094/GroovyInjection.qhelp

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>
+Apache Groovy is a powerful, optionally typed and dynamic language, 
+with static-typing and static compilation capabilities.
+
+It integrates smoothly with any Java program, 
+and immediately delivers to your application powerful features, 
+including scripting capabilities, Domain-Specific Language authoring, 
+runtime and compile-time meta-programming and functional programming.
+
+If a Groovy script is built using attacker-controlled data, 
+and then evaluated, then it may allow the attacker to achieve Remote Code Execution.
+</p>
+</overview>
+
+<recommendation>
+<p>
+It is generally recommended to avoid using untrusted input in a Groovy evaluation.
+If it is not possible, use org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SecureGroovyScript.
+</p>
+</recommendation>
+
+<example>
+<p>
+The following example uses untrusted data to evaluate a Groovy script.
+</p>
+<sample src="GroovyInjection.java" />
+
+</example>
+
+</qhelp>
```

Comment:
Reference worth pointing to:
https://blog.orange.tw/2019/02/abusing-meta-programming-for-unauthenticated-rce.html